src/Pages/Profile/Addresses/Addresses.ts

```diff
@@ -0,0 +1,98 @@
+import Div from '../../../ui-components/Div/Div';
+import Button from '../../../ui-components/Button/Button';
+import api from '../../../api/Api';
+import userState from '../../../states/UserState';
+import { AddressItem } from './AddressItem';
+import { IAddressData } from '../../../interfaces/interfaces';
+import { profile } from '../../../data/data';
+import Span from '../../../ui-components/Span/Span';
+
+class Addresses {
+  private container: Div;
+  private addressContainer: Div;
+  private addNewAddressButton: Button;
+  private addressTitle: Span;
+
+  constructor() {
+    this.container = new Div('profile__addresses');
+    this.addressTitle = new Span(profile.addressesTitle, 'addresses__title', this.container.get());
+    this.addressContainer = new Div('addresses__addresses-container', this.container.get());
+    this.addNewAddressButton = new Button('Add new address', 'addresses__btn', this.container.get());
+  }
+
+  get() {
+    return this.container.get();
+  }
+
+  render(parentElement?: HTMLElement) {
+    this.updateContent();
+    if (parentElement) parentElement.append(this.container.get());
+    this.addNewAddressButton.get().addEventListener('click', async () => {
+      await this.addNewAddress();
+    });
+  }
+
+  updateContent() {
+    this.addressContainer.get().innerHTML = '';
+    this.showAddresses();
+  }
+
+  async showAddresses() {
+    if (userState.getUserId() !== null) {
+      await api.getCustomerById(userState.getUserId() as string).then((data) => {
+        const addresses = data?.data.addresses;
+        let defaultBillingAddressId = '';
+        let defaultShippingAddressId = '';
+        if (data?.data.defaultBillingAddressId) defaultBillingAddressId = data?.data.defaultBillingAddressId;
+        if (data?.data.defaultShippingAddressId) defaultShippingAddressId = data?.data.defaultShippingAddressId;
```

Comment:
to prevent using  data?.data construction, you should use condition.
But what if data will be null?